src/models/ShowPreference.py

```diff
@@ -0,0 +1,133 @@
+
+    def __init__(self, api_id: int, pict: str, next_ep_num: int, next_ep_date: int,
+                 next_season_num: date, title: str, new_ep_acknoweldged: int):
+        self.__set_api_id(api_id)
+        self.__set_pict(pict)
+        self.__set_next_ep_num(next_ep_num)
+        self.__set_next_ep_date(next_ep_date)
+        self.__set_next_season_num(next_season_num)
+        self.__set_title(title)
+        self.__set_new_ep_acknoweldged(new_ep_acknoweldged)
+
+    @property
+    def api_id(self):
+        return self.__api_id
+
+    def __set_api_id(self, api_id: int):
+        if type(api_id) is not int and api_id is not None:
+            raise TypeError("api_id should be an int")
+        else:
+            self.__api_id = api_id
+
+    @property
+    def pict(self):
+        return self.__pict
+
+    def __set_pict(self, pict: str):
+        if type(pict) is not str and pict is not None:
+            raise TypeError("pict should be an string")
+        else:
+            self.__pict = pict
+
+    @property
+    def next_ep_num(self):
+        return self.__next_ep_num
+
+    def __set_next_ep_num(self, next_ep_num: int):
+        if type(next_ep_num) is not int and next_ep_num is not None:
+            raise TypeError("next_ep_num should be an integer")
+        else:
+            self.__next_ep_num = next_ep_num
+
+    @property
+    def next_ep_date(self):
+        return self.__next_ep_date
+
+    def __set_next_ep_date(self, next_ep_date: str):
+        if type(next_ep_date) is not str and next_ep_date is not None:
+            raise TypeError("next_ep_date should be an string")
+        else:
+            self.__next_ep_date = next_ep_date
+
+    @property
+    def next_season_num(self):
+        return self.__next_season_num
+
+    def __set_next_season_num(self, next_season_num: int):
+        if type(next_season_num) is not int and next_season_num is not None:
+            raise TypeError("next_season_num should be an integer")
+        else:
+            self.__next_season_num = next_season_num
+
+    @property
+    def title(self):
+        return self.__title
+
+    def __set_title(self, title: str):
+        if type(title) is not str and title is not None:
+            raise TypeError("title should be an string")
+        else:
+            self.__title = title
+
+    @property
+    def new_ep_acknoweldged(self):
+        return self.__new_ep_acknoweldged
+
+    def __set_new_ep_acknoweldged(self, new_ep_acknoweldged: int):
+        if type(new_ep_acknoweldged) is not int and new_ep_acknoweldged is not None:
+            raise TypeError("new_ep_acknoweldged should be an integer")
+        else:
+            self.__new_ep_acknoweldged = new_ep_acknoweldged
+
+    @classmethod
+    def get_show_preferences_from_user(cls, user: User, my_db: MyDBConnection):
+        show_pref_db_res = my_db.exec_one("""
+            SELECT $rows FROM show 
```

Comment:
As I have never seen a sql request with a "$", I just ask if it is not a typo

Reply:
It's just replaced by the list of column we want to retrieves, should be $cols actually :p (modified)